docs/tutorial/06-marketplace-compose.mdx

```diff
@@ -605,20 +797,22 @@ If we wanted to build a central marketplace on-chain, we could use a contract th
 // references so that anyone can access them
 pub contract Marketplace {
     // Data structure to store active sales
-    pub var tokensForSale: [Capability<&SaleCollection>]
+    pub var tokensForSale: {Address: Capability<&SaleCollection>)}
 
     // listSaleCollection lists a users sale reference in the array
     // and returns the index of the sale so that users can know
     // how to remove it from the marketplace
-    pub fun listSaleCollection(collection: Capability<&SaleCollection>): Int {
-        self.tokensForSale.append(collection)
-        return (self.tokensForSale.length - 1)
+    pub fun listSaleCollection(collection: Capability<&SaleCollection>) {
+        let saleRef = collection.borrow()
+            ?? panic("Invalid sale collection capability")
+
+        self.tokensForSale[saleRef.owner?.address] = collection
```

Comment:
Does this type-check? Indexing using an optional into the dictionary shouldn't be possible. Maybe you meant
```suggestion
        self.tokensForSale[saleRef.owner!.address] = collection
```

Reply:
that code isn't meant to actually be used, just a little example, but that is a good point. I fixed the line. :)